crates/uv/tests/lock.rs

```diff
@@ -2646,3 +2646,189 @@ fn relative_and_absolute_paths() -> Result<()> {
 
     Ok(())
 }
+
+/// Lock a project that includes cyclic dependencies.
+#[test]
+fn lock_cycles() -> Result<()> {
+    let context = TestContext::new("3.12");
+
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(
+        r#"
+        [project]
+        name = "project"
+        version = "0.1.0"
+        requires-python = ">=3.12"
+        dependencies = ["testtools==2.3.0", "fixtures==3.0.0"]
```

Comment:
Smallest case I could find (surfaced from an old Poetry issue and then digging through the versions that were relevant back in 2018)...